matrix/Matrix.hpp

```diff
@@ -499,8 +499,8 @@ bool isEqual(const Matrix<Type, M, N> &x,
 
 
     if (!equal) {
-        char buf_x[100];
-        char buf_y[100];
+        char * buf_x = new char[100];
```

Comment:
Why is it then not dynamic here? We already generate the char array on the heap for print.
The easiest solution could be something like:
```
printf("not equal:\n x:\n");
x.print();
printf("\ny:\n");
y.print();
```